conda/activate.py

```diff
@@ -39,8 +39,21 @@ class Activator(object):
     # To implement support for a new shell, ideally one would only need to add shell-specific
     # information to the __init__ method of this class.
 
-    def __init__(self, shell, arguments=None):
-        self.shell = shell
+    # The following instance variables must be defined by each implementation.
+    shell = None
+    pathsep_join = None
+    path_conversion = None
+    script_extension = None
+    tempfile_extension = None  # None means write instructions to stdout rather than a temp file
+    shift_args = None
+    command_join = None
+
+    unset_var_tmpl = None
```

Comment:
We could replace those template with methods that return the appropriate formatted strings, like suggested in https://github.com/conda/conda/pull/6585#issuecomment-354178822, which would make the code a little cleaner/better readable.
But that could also be done in a subsequent PR anytime and is not crucial for this one.